Framework/src/RRNG.cc

```diff
@@ -0,0 +1,131 @@
+#include "PandaTree/Framework/interface/RRNG.h"
+#include "Math/QuantFuncMathCore.h"
+#include "TString.h"
+#include <iostream>
+#include <stdexcept>
+#include <cassert>
+
+using namespace std;
+using namespace panda;
+
+int RRNG::_default_idx;
+
+RRNG::RRNG(int seed, int maxEntropy): 
+  _rng(seed),
+  _maxEntropy(maxEntropy)
+{
+  assert(_maxEntropy > 0);
+  _default_idx = -1;
+  _x = new double[_maxEntropy];
+  _available = new bool[_maxEntropy];
+  generate();
+}
+
+RRNG::RRNG(RRNG const& other): 
+  _rng(other._rng),
+  _maxEntropy(other._maxEntropy)
+{
+  _x = new double[_maxEntropy];
+  _available = new bool[_maxEntropy];
+  generate();
+}
+
+void RRNG::generate() 
+{
+  _rng.RndmArray(_maxEntropy, _x);
```

Comment:
Oh, and if we go the route of resizeable vectors, we can put a check here for size. If it's zero, set it to some small default number, like 10, before generating. That way, the code doesn't unhelpfully seg fault if someone forgets to call (or doesn't feel like calling) `initRNG()`

Reply:
I'm confused. `initRNG` creates the object. Regardless of whether there's a size check here, there's going to be a seg fault if you try to access `event.rng` without `event.initRNG`. And the constructor for `RRNG` requires a size. I suppose I can throw an error if the constructor is passed 0, but that really seems like we're babysitting the user...

Reply:
I realize the resizing would be better in the access method.

We shouldn't require our level of knowledge of the EventBase header just to use this. If we're assuming the use case is that small, there's no point in making a central generator.

Reply:
I'm not sure what you're suggesting here, and what it has to do with the generate function. I don't think a user needs detailed knowledge of the headers to know that initializing an array of random numbers of 0 size is going to produce 0 random numbers.

Reply:
Yeah, I think this is the wrong line to make the suggestion. Let me find better lines.